platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1560,34 +1587,29 @@ namespace platoon_strategic_ihp
                     ROS_DEBUG_STREAM("Change to LeaderAborting state and waiting for " << msg.header.sender_id << " to join as the new platoon leader");
 
                     // ----------------- give up leader position and look for new leader --------------------------
-                    ROS_DEBUG_STREAM("Received positive response for plan id = " << pm_.current_plan.planId);
-                    ROS_DEBUG_STREAM("Change to LEADERABORTING state and notify trajectory failure in order to replan");
 
                     // adjust for frontal join. Platoon info is related to the platoon at back of the candidate leader vehicle.
+                    // Don't want an action plan here
                     pm_.current_platoon_state = PlatoonState::LEADERABORTING;
                     candidatestateStartTime = ros::Time::now().toNSec() / 1000000;
 
-                    // change platoon ID to candidate leader --> read it from current platoonplan, plan is send out by frontal joiner
-                    targetPlatoonId = potentialNewPlatoonId_front_;
-                    ROS_DEBUG_STREAM("targetPlatoonId = " << targetPlatoonId);
-                    // change leader to candidate leader --> read it from current platoonplan --> peerID is the frontal joiner vehID 
-                    pm_.targetLeaderId = pm_.current_plan.peerId;
-                    ROS_DEBUG_STREAM("pm_.targetLeaderId = " << pm_.targetLeaderId);
-                    // -----------------------------------------------------------------------------------------
-                    
-                    waitingStartTime = ros::Time::now().toNSec() / 1000000;
+                    // If we are testing with a single vehicle representing this platoon, then we don't yet have a platoon ID,
```

Comment:
In mob_op_cb_leader() around line 1063 it decides whether to use the existing platoon's ID, if one exists, or to propose a new one.  The logic here is supposed to complement that so that both sides of the conversation agree on where the platoon ID is coming from.  If the host (old leader) is in a real platoon already, we should keep its ID.  Therefore, I believe it is good as is.